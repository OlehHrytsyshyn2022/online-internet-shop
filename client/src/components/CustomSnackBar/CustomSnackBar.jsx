import React from "react";
import { Snackbar, Alert, AlertTitle } from "@mui/material";
import CheckCircleIcon from "@mui/icons-material/CheckCircle";
import ErrorIcon from "@mui/icons-material/Error";
import InfoIcon from "@mui/icons-material/Info";

const CustomSnackbar = ({ open, onClose, text, titleText, severity }) => {
  const customIconMapping = {
    success: (
      <CheckCircleIcon fontSize="inherit" style={{ color: "#A18A68" }} />
    ),
    error: <ErrorIcon fontSize="inherit" style={{ color: "#A18A68" }} />,
    info: <InfoIcon fontSize="inherit" style={{ color: "#A18A68" }} />,
  };

  return (
    <Snackbar
      anchorOrigin={{ vertical: "top", horizontal: "center" }}
      open={open}
      autoHideDuration={3000}
      onClose={onClose}
      sx={{ marginTop: "50px" }}
    >
      <Alert
        onClose={onClose}
        severity={severity}
        style={{ color: "#A18A68", backgroundColor: "#EFEFEF" }}
        iconMapping={customIconMapping}
      >
        <AlertTitle>{titleText}</AlertTitle>
        {text}
      </Alert>
    </Snackbar>
  );
};

export default CustomSnackbar;
