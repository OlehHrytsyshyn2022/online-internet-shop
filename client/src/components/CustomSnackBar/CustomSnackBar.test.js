import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import CustomSnackBar from "./CustomSnackBar";

describe("CustomSnackbar", () => {
  test("renders CustomSnackBar component", () => {
    const onClose = jest.fn();
    const { getByText } = render(
      <CustomSnackBar
        open={true}
        onClose={onClose}
        text="This is a test message"
        titleText="Test Title"
        severity="success"
      />
    );
    const snackbarText = getByText("This is a test message");
    expect(snackbarText).toBeInTheDocument();
  });

  test("Snapshot Test CustomSnackBar", () => {
    const onClose = jest.fn();
    const { container } = render(
      <CustomSnackBar
        open={true}
        onClose={onClose}
        text="This is a test message"
        titleText="Test Title"
        severity="success"
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("calls onClick handler when clicked", () => {
    const onClickMock = jest.fn();
    render(
      <CustomSnackBar
        open={true}
        onClose={onClickMock}
        text="This is a test message"
        titleText="Test Title"
        severity="success"
      />
    );
    fireEvent.click(screen.getByLabelText("Close"));
    expect(onClickMock).toHaveBeenCalled();
  });
});
