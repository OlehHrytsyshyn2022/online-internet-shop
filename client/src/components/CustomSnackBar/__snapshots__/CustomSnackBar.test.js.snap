// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSnackbar Snapshot Test CustomSnackBar 1`] = `
<div
  class="MuiSnackbar-root MuiSnackbar-anchorOriginTopCenter css-154p0rq-MuiSnackbar-root"
  role="presentation"
>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardSuccess MuiAlert-standard css-wnkddu-MuiPaper-root-MuiAlert-root"
    direction="down"
    role="alert"
    style="opacity: 1; transform: scale(1, 1); color: rgb(161, 138, 104); background-color: rgb(239, 239, 239); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
        data-testid="CheckCircleIcon"
        focusable="false"
        style="color: rgb(161, 138, 104);"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message css-1pxa9xg-MuiAlert-message"
    >
      <div
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
      >
        Test Title
      </div>
      This is a test message
    </div>
    <div
      class="MuiAlert-action css-ki1hdl-MuiAlert-action"
    >
      <button
        aria-label="Close"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall css-1e0d89p-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        title="Close"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CloseIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
